crux_core/src/capabilities/render.rs

```diff
@@ -35,34 +37,27 @@ impl Operation for RenderOperation {
 }
 
 /// Public API of the capability, called by App::update.
-impl<Ev> Render<Ev>
-where
-    Ev: 'static,
-{
-    pub fn new(context: CapabilityContext<RenderOperation, Ev>) -> Self {
+impl Render {
+    pub fn new(context: CapabilityContext<RenderOperation>) -> Self {
         Self { context }
     }
 
     /// Call `render` from [`App::update`](crate::App::update) to signal to the Shell that
     /// UI should be re-drawn.
-    pub fn render(&self) {
+    pub fn render<Event>(&self) -> Command<Event> {
+        Command::empty_effect(self.render_async())
+    }
+
+    /// Call `render` from [`App::update`](crate::App::update) to signal to the Shell that
+    /// UI should be re-drawn.
+    pub fn render_async(&self) -> impl Future<Output = ()> {
         let ctx = self.context.clone();
-        self.context.spawn(async move {
+        async move {
             ctx.notify_shell(RenderOperation).await;
-        });
+        }
```

Comment:
Thinking about it a bit more - the problem with that idea is that those special futures will be inside one of the states of an anonymous future made by an `async` block, which is opaque, isn't it... the "innermost" future needs _some_ way of exposing the operation that can resolve it to the outside.

Reply:
But prehaps this could be done on a more local scale within the `Command`. The command could take a function which gets an equivalent of the context provided by the executor later. That context would just have a local channel back up to the command, allowing to bypass the anonymous future layers.... or something along those lines.